ui/src/views/Provisioners/ViewWorker/index.jsx

```diff
@@ -151,6 +157,49 @@ export default class ViewWorker extends Component {
     });
   }
 
+  handleTerminateDialogActionOpen = (workerId, workerGroup, workerPoolId) => {
+    this.setState({
+      terminateDialogOpen: true,
+      terminateDialogTitle: 'Terminate Worker?',
+      terminateDialogBody: `This will terminate the worker with id ${workerId} in group ${workerGroup} within worker pool ${workerPoolId}.`,
+      terminateDialogConfirmText: 'Terminate Worker',
+      workerPoolId,
+      workerGroup,
+      workerId,
+    });
+  };
+
+  handleTerminateDeleteClick = async () => {
+    const { workerPoolId, workerGroup, workerId } = this.state;
+    const { user } = this.props;
+
+    this.setState({
+      terminateDialogError: null,
+    });
+
+    try {
+      removeWorker({ workerPoolId, workerGroup, workerId, user });
```

Comment:
this should very likely be `await removeWorker` unless you want to to run in "background", in which case try-catch is useless

Reply:
Great catch, thank you!